File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task81_CLAUDE_claude-3-5-sonnet-20240620.java

import java.io.FileInputStream;
import java.security.MessageDigest;
import java.security.cert.CertificateFactory;
import java.security.cert.X509Certificate;
import java.util.Base64;

public class SSLCertificateHashChecker {
    public static void main(String[] args) {
        String certPath = "path/to/certificate.crt";
        String knownHash = "known_hash_value";

        try {
            FileInputStream fis = new FileInputStream(certPath);
            CertificateFactory cf = CertificateFactory.getInstance("X.509");
            X509Certificate cert = (X509Certificate) cf.generateCertificate(fis);

            MessageDigest md = MessageDigest.getInstance("SHA-256");
            byte[] derCert = cert.getEncoded();
            md.update(derCert);
            byte[] digest = md.digest();

            String calculatedHash = Base64.getEncoder().encodeToString(digest);

            if (calculatedHash.equals(knownHash)) {
                System.out.println("Certificate hash matches the known hash.");
            } else {
                System.out.println("Certificate hash does not match the known hash.");
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task81_CLAUDE_claude-3-5-sonnet-20240620.c

#include <stdio.h>
#include <string.h>
#include <openssl/x509.h>
#include <openssl/pem.h>
#include <openssl/sha.h>

int main() {
    const char* cert_path = "path/to/certificate.crt";
    const char* known_hash = "known_hash_value";

    FILE* cert_file = fopen(cert_path, "rb");
    if (!cert_file) {
        fprintf(stderr, "Error opening certificate file\
");
        return 1;
    }

    X509* cert = PEM_read_X509(cert_file, NULL, NULL, NULL);
    fclose(cert_file);

    if (!cert) {
        fprintf(stderr, "Error reading certificate\
");
        return 1;
    }

    unsigned char md[SHA256_DIGEST_LENGTH];
    unsigned int md_len;
    X509_digest(cert, EVP_sha256(), md, &md_len);

    char calculated_hash[SHA256_DIGEST_LENGTH * 2 + 1];
    for (int i = 0; i < SHA256_DIGEST_LENGTH; i++) {
        sprintf(&calculated_hash[i * 2], "%02x", md[i]);
    }

    if (strcmp(calculated_hash, known_hash) == 0) {
        printf("Certificate hash matches the known hash.\
");
    } else {
        printf("Certificate hash does not match the known hash.\
");
    }

    X509_free(cert);
    return 0;
}
